### PR TITLE
Animation updates

### PR DESCRIPTION
**Pull request by Melody-Ruth**
We use the walking animation if the player is moving horizontally, the jumping animation if they are in the air and aren't moving horizontally, and the idle animation if neither of those things are true. I can adjust these if people want to do something else (e.g. I could prioritize the jumping animation over the walking animation instead of vice versa).

### CODE REVIEW

**Review by Alanine42 — APPROVED** [review]
I'm down for smaller PRs

**Review by realmatthewpeng — APPROVED** [review]
Works on my machine!

File: server/src/bge/EventHandler.cpp [hunk 1]
```diff
@@ -176,8 +176,12 @@ namespace bge {
 		VelocityComponent& velTop = velocityCM->lookup(top);
 		VelocityComponent& velBottom = velocityCM->lookup(bottom);
 		velTop.velocity.y = std::max({velBottom.velocity.y, velTop.velocity.y, std::abs(yOverlapDistance)});
-
 		// fix: prevent box clipping by moving the top player up by at least their overlap distance ^
+		
+		// make sure top player displays idle animation if not moving
+		if (length(velTop.velocity) < 0.01f) {
```
Comment (benjxia) [inline server/src/bge/EventHandler.cpp:182]: nit: glm::length

**Review by benjxia — APPROVED** [review]

**Review by Metarus — APPROVED** [review]
works on my machine, lgtm

**Pull request merged**